packages/host/app/components/operator-mode/profile-settings-modal.gts

```diff
@@ -1,128 +1,137 @@
-    await new Promise((resolve) => setTimeout(resolve, 2000));
-    this.saveSuccessIndicatorShown = false;
-  });
-
-  private setInitialValues = restartableTask(async () => {
-    await this.matrixService.profile.loaded;
-    this.displayName = this.matrixService.profile.displayName;
-  });
-
-  @action private onSubmit(event: Event) {
-    event.preventDefault();
-    this.saveTask.perform();
-  }
+class PasswordModal extends Component<PasswordModalSignature> {
+  @tracked private password = '';
 
-  private get saveButtonText() {
-    if (this.saveSuccessIndicatorShown) return 'Saved!';
-    return this.saveTask.isRunning ? 'Saving…' : 'Save';
+  private get passwordInputState() {
+    return this.args.passwordError ? 'invalid' : 'initial';
   }
 
-  private get isDisplayNameValid() {
-    return this.displayName !== undefined && this.displayName.length > 0;
-  }
-
-  private get isSaveButtonDisabled() {
-    return (
-      this.saveTask.isRunning ||
-      !this.isDisplayNameValid ||
-      this.displayName === this.matrixService.profile.displayName
-    );
-  }
-
-  constructor(owner: unknown, args: any) {
-    super(owner, args);
-    this.setInitialValues.perform();
+  @action private setPassword(password: string) {
+    this.args.clearPasswordError();
+    this.password = password;
   }
 
   <template>
+    <ModalContainer
+      @onClose={{@togglePasswordModal}}
+      @title='Confirm Identity'
+      @size='small'
+      @centered={{true}}
+      @isOpen={{true}}
+      class='password-modal'
+      data-test-password-modal
+    >
+      <:content>
+        <div class='instructions'>
+          Confirm your identity by entering your password below.
+        </div>
+        <FieldContainer @tag='label' @vertical={{true}}>
+          <BoxelInput
+            data-test-password-field
+            @type='password'
+            @value={{this.password}}
+            @state={{this.passwordInputState}}
+            @errorMessage={{@passwordError}}
+            @onInput={{this.setPassword}}
+          />
+        </FieldContainer>
+      </:content>
+      <:footer>
+        <div class='buttons'>
+          <BoxelButton
+            data-test-confirm-cancel-button
+            @size='tall'
+            @kind='secondary-light'
+            {{on 'click' @togglePasswordModal}}
+          >
+            Cancel
+          </BoxelButton>
+
+          <BoxelButton
+            @kind='primary'
+            @size='tall'
+            @disabled={{not this.password}}
+            class='confirm-button'
+            {{on 'click' (fn @confirmPassword this.password)}}
+            data-test-confirm-password-button
+          >
+            Confirm
+          </BoxelButton>
+        </div>
+      </:footer>
+    </ModalContainer>
     <style>
+      .password-modal :deep(.boxel-modal__inner) {
+        height: 21rem;
+        margin-top: calc((100vh - 21rem) / 2);
+      }
+      .password-modal :deep(.invalid) {
+        box-shadow: none;
```

Comment:
there seems to be a box shadow now on the invalid indicator that looks really bad with the `BoxelInput` component. perhaps this was an unintentional side effect?